iFixit/react-commerce
# Add bulk operations to addons plugin

closes #266 

This PR adds bulk operations to the Strapi addons plugin. As part of this the addons plugin UI has been updated with a navigation sidebar.
This PR introduces also some changes to Strapi Typescript config that are necessary for zod to work correctly and to improve types overall.

## QA

1. Open the [Strapi preview instance](https://add-bulk-operations.govinor.com/admin/plugins/addons/bulk-operations) for this branch
2. Navigate to the addon plugin and click "Bulk operations" on the secondary nav bar
3. Try to export a collection type (e.g. "product list"): this should give you a link to download the CSV
    <img width="1400" alt="Screenshot 2023-07-05 at 22 12 21" src="https://github.com/iFixit/react-commerce/assets/4640135/ccce0271-adc8-4d1b-a0d6-1a3a36be4874">

4. Open the downloaded CSV with Google Spreadsheet and update the entries value. Add also new entries (new entries are rows with the id column left blank) and you can delete the rows that you don't need to update.
5. Export the updated file as a CSV
6. In the addons plugin "Bulk Operations" page, click on the "import" tab and follow the instructions to upload the file and start the import
    <img width="1667" alt="Screenshot 2023-07-05 at 22 13 17" src="https://github.com/iFixit/react-commerce/assets/4640135/94ecbb59-a46e-44b4-a2e2-c7deb86d062d">

7. Once the import finishes, navigate to the Strapi Content Manager and verify that the changes you've made to the CSV have been applied to the collections
